includes/abstracts/abstract.llms.post.model.php

```diff
@@ -910,70 +911,130 @@ protected function scrub_field( $val, $type ) {
+				$key_or_array => $val,
+			);
+		} else {
+			$model_array = $key_or_array;
+		}
+		return $this->update_model( $model_array );
 
-		// update WordPress Post Properties using the wp_insert_post() function
-		if ( in_array( $key, array_keys( $this->get_post_properties() ) ) ) {
+	}
 
-			$post_key = 'post_' . $key;
 
-			switch ( $key ) {
+	/**
+	 * Updater
+	 *
+	 * @since [version]
+	 *
+	 * @param array $model_array Associative array of key/val pairs.
+	 * @return boolean true on success, false on error or if there was nothing to update.
+	 */
+	public function update_model( $model_array ) {
 
-				case 'content':
-					$val = apply_filters( 'content_save_pre', $val );
-				break;
+		if ( empty( $model_array ) ) {
+			return false;
+		}
 
-				case 'excerpt':
-					$val = apply_filters( 'excerpt_save_pre', $val );
-				break;
+		$llms_post = array(
+			'post' => array(),
+			'meta' => array(),
+		);
 
-				case 'menu_order':
-					$post_key = 'menu_order';
-				break;
+		$post_properties = array_keys( $this->get_post_properties() );
+		$unsettable_properties = $this->get_unsettable_properties();
 
-				case 'title':
-					$val = apply_filters( 'title_save_pre', $val );
-				break;
+		foreach ( $model_array as $key => $val ) {
 
-			}
+			$val = $this->scrub( $key, $val );
 
-			$args = array(
-				'ID' => $this->get( 'id' ),
-			);
+			// update WordPress Post Properties using the wp_insert_post() function
+			if ( in_array( $key, $post_properties ) ) {
 
-			$args[ $post_key ] = apply_filters( 'llms_set_' . $this->model_post_type . '_' . $key, $val, $this );
+				$type           = 'post';
+				$llms_post_key  = "post_{$key}";
 
-			if ( wp_update_post( wp_slash( $args ) ) ) {
-				$this->post->{$post_key} = $val;
-				return true;
-			} else {
-				return false;
-			}
-		} // End if().
-		elseif ( ! in_array( $key, $this->get_unsettable_properties() ) ) {
+				switch ( $key ) {
+
+					case 'content':
+						$val = apply_filters( 'content_save_pre', $val );
+					break;
 
-			$u = update_post_meta( $this->id, $this->meta_prefix . $key, apply_filters( 'llms_set_' . $this->model_post_type . '_' . $key, $val, $this ) );
-			if ( is_numeric( $u ) || true === $u ) {
-				return true;
+					case 'excerpt':
+						$val = apply_filters( 'excerpt_save_pre', $val );
+					break;
+
+					case 'menu_order':
+						$llms_post_key = 'menu_order';
+					break;
+
+					case 'title':
+						$val = apply_filters( 'title_save_pre', $val );
+					break;
+				}
+			} elseif ( ! in_array( $key, $unsettable_properties ) ) {
+				$type          = 'meta';
+				$llms_post_key = $key;
 			} else {
-				return false;
+				continue;
 			}
-		} // we have a problem...
-		else {
 
+			$llms_post[ $type ][ $llms_post_key ] = apply_filters( 'llms_set_' . $this->model_post_type . '_' . $key, $val, $this );
+
+		}// End foreach().
+
+		if ( empty( $llms_post['post'] ) && empty( $llms_post['meta'] ) ) {
 			return false;
+		}
+
+		$has_error = false;
```

Comment:
this can become a WP_Error object, collect the errors and later check if there are any errors to and return either the object OR `false` if `wp_error` is `false`

Reply:
in 565950b